### ISSUE TITLE
AI: Balancing of accepting "unfavorable" trades (reason in this example related to gifted gold)

### ISSUE THREAD

**Issue by CrsiX** [yairm210/Unciv#11969]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Game Version

4.12.8 Build 1019 on Android

### Describe the bug

Even though it is vastly stronger in military power, economic strength and population size and therefore not reliant on anyone else in the game, even though it additionally has a few city states as its allies, the AI Arabia accepts an extremely unfavorable trade with player Germany:

Germany (player) gives:
- Defensive Pact
- Declare War against Polynesia

Arabia (AI) gives:
- Defensive Pact
- Cotton 1
- Gems 1
- Incense 1
- Spices 1
- Wine 1
- Citrus 1
- Copper 1
- Khurasan (city! 10 inhabitants)
- Medina (city! 16 inhabitants)

See the screenshots or the saved game file for how big the difference between the two is. Note that Polynesia (which Arabia just declared its war against), is extremely weak. Also note that the fourth major civ in the game was annihilated by Arabia less than 10 turns ago. As a player, the only sane option is preparing for the worst. But when Arabia offered the player to join its war, the player asked "why should I?" and maxed out the demands as seen in the save game.

**TL;DR:**

Balancing is very hard. But it goes against all reason that Arabia accepts this trade offer. This should offer an example of a situation, which should _not_ be accepted by Arabia.

**Suspicion**

I have some thoughts about it, see comment below.

### Steps to Reproduce

1. Explore the map and the victory status -- see that Arabia is overwhelmingly strong
2. See the trade that Germany (player) proposed to Arabia (AI)
3. Click next turn
4. Arabia accepted the trade (and someone declared war on the player, but that's not the point)

### Screenshots

<img width="1470" alt="Screenshot1" src="https://github.com/user-attachments/assets/641c285a-1e98-45d3-8e46-4e34b6da5242">
<img width="1470" alt="Screenshot2" src="https://github.com/user-attachments/assets/a8e55680-5161-4399-a30d-036ba61a5025">

### Link to save file

600kB JSON:
https://gist.github.com/CrsiX/13ac755f8ff0f8420b993b8ab9b316c6

### Operating System

Android

### Additional Information

Since I was debugging it on Desktop, it is also reproducible on latest master branch. My thoughts on this issue are in comments below.

**Comment by CrsiX** [comment]
In the [TradeEvaluation.kt](https://github.com/yairm210/Unciv/blob/master/core/src/com/unciv/logic/trade/TradeEvaluation.kt) file, I noticed the following points from the perspective of
the AI player Arabia (`civInfo=Arabia`):

1. In `getTradeAcceptability`, the `sumOfTheirOffers` for evaluation by  Arabia
  gives 11866 points for the two mere offeres (declaration of war &
  defensive pact), while the `sumOfOurOffers` is 17241 which includes two cities
  and various luxuries. The difference is remarkably small, would you not expect
  something like 1:15 as a ratio here? I did not dig further here, though.
2. In `getTradeAcceptability`, the `sumOfOurOffers` for evaluation by Germany
  gives 116552 points,  which seems **way** more than all the other values I
  have seen so far. (This situation happens because upon accepting from Arabia,
  Germany also calls this function during handling.)
3. The `isTradeAcceptable` returns true only because the gifted gold is apparently
  over 5800 which evens out this great difference, so that Arabia accepts the trade.
  (See point 1.) Without that, it would have not been accepted.

After I saw the gifted gold, I checked the following:
repeatedly requesting one new city per turn.
This worked. Aka, when using the save game, one could accept that trade and then in
each following turn ask Arabia for one more city. Since barely any of the cities gets
a higher point value than the 5800 of gifted gold, and this value does not seem
to decrease upon giving away entire cities (it increased slightly instead), I could basically
ask for almost all cities of Arabia. (I tested it, it works. I receive happiness penalty
of doom, though.)

What is the gifted gold used for? Where does it come from?
I am referencing [this](https://github.com/yairm210/Unciv/blob/a50982e9bb58cf195fc3636449db4516d83c80f7/core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt#L722).
The problem is that it is the total of the diplomatic gifts a civ received
from someone else, is it not? Then it must be modified for every
trade, so that this kind of action is not possible, right?

**Comment by CrsiX** [comment]
Proof that the player could ask for new cities in each turn (never with **anything** in return! according to the debugger, it was only accepted because the `gifted gold` was so high -- see [this](https://github.com/yairm210/Unciv/blob/a50982e9bb58cf195fc3636449db4516d83c80f7/core/src/com/unciv/logic/trade/TradeEvaluation.kt#L93) line):

<img width="1470" alt="Screenshot3" src="https://github.com/user-attachments/assets/3821e763-0062-4079-aa7f-e4ee3796285e">

(Note that this resulted in a happiness of -192 and Arabia declared war after losing more than 10 cities.)

**Comment by tuvus** [comment]
Yes, the problem lies in where the gifts are coming from. 
I finally figured out the problem and will make a PR on this soon. It came from missing an extra set of parenthesis.

**Comment by CrsiX** [comment]
Thank you for your quick reply and the quick fix :)

That does not answer one part of my question, though: what is gifted gold? How do you accumulate it? And why does it not decrease (because this leaves open the second half of the issue)?

**Comment by tuvus** [comment]
(Actually it wasn't a quick fix. Other people had reported it at least a week ago, however your save file was enough to let me reproduce it!)

Gifted gold is the gold difference of a trade to an AI. The gold gift is then stored as a diplomatic modifier to slightly boost relations. It is also inflation adjusted. (Although the inflation function could probably be looked at more) Gifted gold can be used as gold credit in trades.
Gifted gold does decrease over time, it looses the max of either 40 gold, or 2%-40% (Ally to Unforgivable) depending on the relations. Maybe the min needs to be made higher though.

In the future, it would be good to have some inefficiencies with gifting. Such as only 80% of the extra gold value is converted to gifts.

**Comment by CrsiX** [comment]
> (Actually it wasn't a quick fix. Other people had reported it at least a week ago, however your save file was enough to let me reproduce it!)

I'm glad I could help you out there :)

> Gifted gold is the gold difference of a trade to an AI

So I accumulated the high amount of gifted gold by trading efficiently with the AI earlier? I did some research agreements, exchanged luxuries and became friendly an such things. Interesting.
I'm still not sure how the value got so high, because that still leaves me wondering: should the gifted gold not be reduced when "used"? As in the screenshot above, I literally asked for whole cities from the AI without anything in return, and it just gave me these cities. Partially/entirely without decreasing the gifted gold modifier (stepped through it with the debugger), I'm not sure about all effects that modify this value.
Maybe the meaning should be changed from "gifted gold" to "balanced gold"? So that it could decrease accordingly?

**Comment by CrsiX** [comment]
(Maybe something for a different issue)
After continuing this game a bit, Arabia destroyed Polynesia. Now only the player was the last other civ left, so Arabia attacked it immediately. This makes perfectly sense given the military strength, but I can tell that the cities of Arabia were neglected (since I got them, see earlier screenshot) - aka they did not have many buildings & infrastructure (besides roads everywhere). Means, that the AI at some point in time (turn ~130?) decided to max out military and wage war. This is a perfectly valid approach, but I feel like it is not the best tactic. If someone could defeat these units, all cities would be way weaker than comparing cities from other civs. Meaning a disadvantage in the long run. What's your thought on that? Or is it just me who wants to have long games and opt for cultural/scientific wins? xD

**Comment by tuvus** [comment]
> So I accumulated the high amount of gifted gold by trading efficiently with the AI earlier? 

Not exactly. You got the gifted gold by accepting a join their war request which was bugged. 
A value of 100 on the gifted gold modifier is equivalent to 100,000 gold.

**Comment by CrsiX** [comment]
> A value of 100 on the gifted gold modifier is equivalent to 100,000 gold.

Dang! Alright, then this makes a bit more sense to me now.

**Comment by tuvus** [comment]
> If someone could defeat these units, all cities would be way weaker than comparing cities from other civs. Meaning a disadvantage in the long run. What's your thought on that? Or is it just me who wants to have long games and opt for cultural/scientific wins? xD

Balancing is very hard, we want the AI to declare some wars but not too many. Which is a very difficult balance to make without some artificial components. The AI is pretty bad at the late game so making it more competent is a priority.

**Comment by CrsiX** [comment]
> Balancing is very hard

Yes absolutely. Sadly, I'm not too into the code to help you out, and giving half-hearted advise from the sidelines isn't helping either. I can provide a late-game save file every now and then if you need :)
